sigle/src/modules/editor/BubbleMenuItems.tsx

```diff
@@ -0,0 +1,195 @@
+import { Editor } from '@tiptap/react';
+import { styled } from '../../stitches.config';
+import {
+  Link1Icon,
+  FontBoldIcon,
+  FontItalicIcon,
+  CodeIcon,
+  Cross2Icon,
+  StrikethroughIcon,
+  UnderlineIcon,
+} from '@radix-ui/react-icons';
+import { forwardRef, useEffect, useImperativeHandle, useState } from 'react';
+import { Flex } from '../../ui';
+
+const BubbleMenuInput = styled('input', {
+  fontSize: '14px',
+  lineHeight: '18px',
+  width: '100%',
+  pl: '$2',
+  pr: '$1',
+  backgroundColor: 'transparent',
+  outline: 'none',
+});
+
+const BubbleMenuButton = styled('button', {
+  color: '$gray12',
+
+  '@xl': {
+    color: '$gray1',
+  },
+
+  variants: {
+    active: {
+      true: {
+        color: '$orange10',
+      },
+    },
+  },
+});
+
+interface ResetLink {
+  resetLink: () => void;
+}
+
+interface BubbleMenuItemsProps {
+  editor: Editor;
+  iconSize?: number;
+}
+
+export const BubbleMenuItems = forwardRef<ResetLink, BubbleMenuItemsProps>(
+  ({ editor, iconSize = 18 }: BubbleMenuItemsProps, ref) => {
+    const [linkState, setLinkState] = useState({ open: false, value: '' });
+
+    useImperativeHandle(ref, () => ({
+      resetLink,
```

Comment:
this along with the forwardRef is allowing the parent component `BubbleMenu` to call the `resetLink()` method when hidden. While this is working, I'm also aware the `useImperativeHandle` hook is said to be avoided in most cases. Do you know if there is a better way to achieve this, or is this use case justified?